src/RemoteResourceS3DecryptController.js

```diff
@@ -98,14 +98,20 @@ module.exports = class RemoteResourceS3DecryptController extends RemoteResourceS
 
     try {
       this.log.debug(`Downloaded from ${source}`);
+      const isCompressed = source.includes('.tar') || source.includes('.tgz');
```

Comment:
This isn't a definitive test to determine if the source is ascii armored or not. Try checking the mimetype on the response. Might have to fall back to just trying one, then checking for the appropriate exception, but that should be the last resort.

Reply:
add isBinary flag based on res.headers['content-type'] === 'binary/octet-stream'. For normal text, COS returns application/octet-stream as the value. Tested against both binary and text rrs3d resources on minikube, download, pgp, uncompress work fine.